src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs

```diff
@@ -131,8 +131,19 @@ PEFile MakePEFile(IAssemblySymbol assembly)
                 // reference assemblies should be fine here, we only need the metadata of references.
                 var reference = _parentCompilation.GetMetadataReference(assembly);
                 Log(CSharpEditorResources.Load_from_0, reference.Display);
-                return TryResolve(reference, PEStreamOptions.PrefetchMetadata)
-                    ?? new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
+
+                var result = TryResolve(reference, PEStreamOptions.PrefetchMetadata);
+                if (result is not null)
+                {
+                    return result;
+                }
+
+                if (File.Exists(reference.Display))
+                {
+                    return new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
```

Comment:
afaict, there's a race condition here as .Exists may succeed, but `new PEFile` may still not find the file.  Do you want to do the safer pattern of "test; try create; catch exception if creation fails"?

Reply:
Normally I'd probably wrap this in IOUtilities.PerformIO and not to the exists check at all, but this is fixing a Prism bug which happily looks at first chance exceptions, so it won't actually make the bug go away. Without a user report, I can only assume this is not a real issue.

Reply:
>  but this is fixing a Prism bug which happily looks at first chance exceptions, so it won't actually make the bug go away.

@davidwengier Hmm? We don't report all first chance exceptions, you can absolutely use PerformIO or something here.

Reply:
Maybe i'm remembering wrong? Maybe it was an RPS or speedometer that counted first chance

Reply:
Ah yes, RPS or speedometer does. Should this use the File.Exists check (which I'm OK with, to be clear) but still wrap it under a PerformIO or catch it explicitly here?